sdk/digitaltwins/Azure.DigitalTwins.Core/tests/TestObjectSerializer.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Serialization;
+
+namespace Azure.DigitalTwins.Core.Tests
+{
+    /// <summary>
+    /// Custom serializer to pass to the DigitalTwins client.
+    /// </summary>
+    public class TestObjectSerializer : ObjectSerializer
+    {
+        private static JsonObjectSerializer _serializer = new JsonObjectSerializer();
+
+        // This field is used by the tests to confirm the function was called.
+        public bool isDeserializeCalled { get; set; }
```

Comment:
`WasDeserializedCalled` - casing and verb tense

Better yet, you might track how many calls were made, and then assert > 0 or a specific number.

Reply:
Sure that is a good idea too. I will change it to int and assert on number of calls.

Reply:
I tried out using an int. Everything works as expected in 'Live' mode. But the 'Record' mode is surely caching something, causing the expected number to continue to increase on every run. For now, I will leave it as bool for this check in but I'll investigate the issue with Azure Test Framework.

Reply:
Looks like I got used to java casing :P